### PR TITLE
feat(api): add notes in diag report v0

### PR DESCRIPTION
**Pull request by Goncharo**
refs. metriport/metriport-internal#1441

### Description

add notes in diag report v0

### CODE REVIEW

File: packages/fhir-converter/src/lib/cda/cda.js [hunk 1]
```diff
@@ -4,9 +4,11 @@
 // -------------------------------------------------------------------------------------------------
 
 let parseString = require("xml2js").parseString;
+let Builder = require("xml2js").Builder;
 let dataHandler = require("../dataHandler/dataHandler");
 const fs = require("fs");
 let minifyXML = require("minify-xml");
+// const { XMLParser, XMLBuilder, XMLValidator } = require("fast-xml-parser");
```
Comment (leite08) [inline packages/fhir-converter/src/lib/cda/cda.js:11]: nit: left-over

File: packages/fhir-converter/src/lib/cda/cda.js [hunk 2]
```diff
@@ -17,10 +19,15 @@ module.exports = class cda extends dataHandler {
...
       for (const key of Object.keys(obj)) {
-        if (key.toLowerCase() == "id" && obj["_"]) {
+        if (key.toLowerCase() == "id") {
           const id = obj[key];
           const idValue = obj["_"];
-          this.idToValueMap[id] = idValue;
+          if (idValue) {
+            this.idToValueMap[id] = idValue;
+          } else {
+            let xmlString = new Builder({ headless: true }).buildObject(obj);
```
Comment (leite08) [inline packages/fhir-converter/src/lib/cda/cda.js:28]: `let` > `const`

File: packages/fhir-converter/src/lib/cda/cda.js [hunk 3]
```diff
@@ -91,9 +109,28 @@ module.exports = class cda extends dataHandler {
           this.findAndReplaceAllReferencesWithTextValues(result);
           result._originalData = minifiedData;
           fulfill(result);
-          // fs.writeFileSync(`../../JSON.json`, JSON.stringify(result, null, 2));
+          fs.writeFileSync(`../../JSON1.json`, JSON.stringify(result, null, 2));
```
Comment (leite08) [inline packages/fhir-converter/src/lib/cda/cda.js:112]: keep this commented out?

File: packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js [hunk 4]
```diff
@@ -586,6 +586,34 @@ module.exports.external = [
...
+    description:
+      "Returns all instances (non-alphanumeric chars replace by '_' in name) of the sections by template id e.g. getFirstCdaSectionsByTemplateId msg '2.16.840.1.113883.10.20.22.2.14' '1.3.6.1.4.1.19376.1.5.3.1.3.1': getFirstCdaSectionsByTemplateId message templateId1 templateId2 …",
+    func: function getFirstCdaSectionsByTemplateId(msg, ...templateIds) {
+      try {
+        var ret = [];
+
+        for (var t = 0; t < templateIds.length - 1; t++) {
+          //-1 because templateIds includes the full message at the end
+          for (var i = 0; i < msg.ClinicalDocument.component.structuredBody.component.length; i++) {
+            let sectionObj = msg.ClinicalDocument.component.structuredBody.component[i].section;
```
Comment (leite08) [inline packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js:600]: let > const

File: packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js [hunk 5]
```diff
@@ -586,6 +586,34 @@ module.exports.external = [
...
+    name: "getAllCdaSectionsByTemplateId",
+    description:
+      "Returns all instances (non-alphanumeric chars replace by '_' in name) of the sections by template id e.g. getFirstCdaSectionsByTemplateId msg '2.16.840.1.113883.10.20.22.2.14' '1.3.6.1.4.1.19376.1.5.3.1.3.1': getFirstCdaSectionsByTemplateId message templateId1 templateId2 …",
+    func: function getFirstCdaSectionsByTemplateId(msg, ...templateIds) {
+      try {
+        var ret = [];
+
+        for (var t = 0; t < templateIds.length - 1; t++) {
+          //-1 because templateIds includes the full message at the end
+          for (var i = 0; i < msg.ClinicalDocument.component.structuredBody.component.length; i++) {
```
Comment (leite08) [inline packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js:599]: Is this comment related to iterating until past the last item? Otherwise, missing `-1` on `msg.ClinicalDocument.component.structuredBody.component.length`

File: packages/fhir-converter/src/templates/cda/Resources/DiagnosticReport.hbs [hunk 6]
```diff
@@ -18,6 +18,12 @@
         },
         
         "effectiveDateTime":"{{formatAsDateTime diagReport.effectiveTime.value}}",
+        {{!-- {{log diagReport}} --}}
```
Comment (leite08) [inline packages/fhir-converter/src/templates/cda/Resources/DiagnosticReport.hbs:21]: left-over

File: packages/fhir-converter/src/templates/cda/Resources/Procedure.hbs [hunk 7]
```diff
@@ -22,9 +22,9 @@
             "performedPeriod": {{>DataType/Period.hbs period=procedureEntry.effectiveTime}},
         {{/if}}
         {{#if procedureEntry.text._}}
-            "annotation": [
-                {{procedureEntry.text._}}",
-            ]
+            "note": {
+                "text": "{{procedureEntry.text._}}",
+            },
```
Comment (leite08) [inline packages/fhir-converter/src/templates/cda/Resources/Procedure.hbs:27]: 🙏

**Review by leite08 — APPROVED** [review]
A few comments, but nothing too serious - only the access beyond the last element of the array

File: packages/fhir-converter/src/lib/cda/cda.js (hunk 1 above)
Comment (Goncharo) [reply packages/fhir-converter/src/lib/cda/cda.js:11]: leaving this in case we need to migrate

File: packages/fhir-converter/src/lib/cda/cda.js (hunk 2 above)
Comment (Goncharo) [reply packages/fhir-converter/src/lib/cda/cda.js:28]: fixed

File: packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js (hunk 5 above)
Comment (Goncharo) [reply packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js:599]: this comment refers to the above if - moved to avoid confusion

File: packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js (hunk 4 above)
Comment (Goncharo) [reply packages/fhir-converter/src/lib/handlebars-converter/handlebars-helpers.js:600]: fixed

File: packages/fhir-converter/src/templates/cda/Resources/DiagnosticReport.hbs (hunk 6 above)
Comment (Goncharo) [reply packages/fhir-converter/src/templates/cda/Resources/DiagnosticReport.hbs]: fixed
